src/helpers/helpers.extras.ts

```diff
@@ -44,61 +42,46 @@ export function throttled(fn, thisArg, updateFn) {
 
 /**
  * Debounces calling `fn` for `delay` ms
- * @param {function} fn - Function to call.
- * @param {number} delay - Delay in ms. 0 = immediate invocation.
- * @returns {function}
  */
-export function debounce(fn, delay) {
+export function debounce<TArgs extends Array<any>>(fn: (...args: TArgs) => void, delay: number) {
   let timeout;
-  return function(...args) {
+  return function(...args: TArgs) {
     if (delay) {
       clearTimeout(timeout);
       timeout = setTimeout(fn, delay, args);
     } else {
-      fn.apply(this, args);
+      fn.apply<any, TArgs, void>(this, args);
     }
     return delay;
   };
 }
 
 /**
  * Converts 'start' to 'left', 'end' to 'right' and others to 'center'
- * @param {string} align start, end, center
  * @private
  */
-export const _toLeftRightCenter = (align) => align === 'start' ? 'left' : align === 'end' ? 'right' : 'center';
+export const _toLeftRightCenter = (align: 'start' | 'end' | 'center') => align === 'start' ? 'left' : align === 'end' ? 'right' : 'center';
 
 /**
  * Returns `start`, `end` or `(start + end) / 2` depending on `align`. Defaults to `center`
- * @param {string} align start, end, center
- * @param {number} start value for start
- * @param {number} end value for end
  * @private
  */
-export const _alignStartEnd = (align, start, end) => align === 'start' ? start : align === 'end' ? end : (start + end) / 2;
+export const _alignStartEnd = (align: 'start' | 'end' | 'center', start: number, end: number) => align === 'start' ? start : align === 'end' ? end : (start + end) / 2;
 
 /**
  * Returns `left`, `right` or `(left + right) / 2` depending on `align`. Defaults to `left`
- * @param {string} align start, end, center
- * @param {number} left value for start
- * @param {number} right value for end
- * @param {boolean} rtl Is this an RTL draw
  * @private
  */
-export const _textX = (align, left, right, rtl) => {
+export const _textX = (align: 'left' | 'right' | 'center', left: number, right: number, rtl: boolean) => {
   const check = rtl ? 'left' : 'right';
   return align === check ? right : align === 'center' ? (left + right) / 2 : left;
 };
 
 /**
  * Return start and count of visible points.
- * @param {object} meta - dataset meta.
- * @param {array} points - array of point elements.
- * @param {boolean} animationsDisabled - if true animation is disabled.
- * @returns {{start: number; count: number}}
  * @private
  */
-export function _getStartAndCountOfVisiblePoints(meta, points, animationsDisabled) {
+export function _getStartAndCountOfVisiblePoints(meta: ChartMeta<'line'>, points: PointElement[], animationsDisabled: boolean) {
```

Comment:
Won't this break if you try to pass a bar metaset to it?

Reply:
I think it will. This function is only used in the line and scatter charts. I should update this to `ChartMeta<'line' | 'scatter'>`